Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoUpdateLink.cpp

```diff
@@ -78,15 +89,26 @@ namespace AzToolsFramework
 
             // Remove the link ids if present in the DOMs. We don't want any overrides to be created on top of linkIds because
             // linkIds are not persistent and will be created dynamically when prefabs are loaded into the editor.
-            instanceDom.RemoveMember(PrefabDomUtils::LinkIdName);
-            sourceDom.RemoveMember(PrefabDomUtils::LinkIdName);
+            if (instanceDom.HasMember(PrefabDomUtils::LinkIdName))
+            {
+                AZ_Warning("Prefab", false, "Linked instance DOM from link with id '%llu' shouldn't contains key 'LinkId'",
```

Comment:
This needs to be removed. The nested instance dom will always have a linkId because it is designed that way